ffi/ffi.cabal

```diff
@@ -34,10 +54,6 @@ executable ffi-test
                      , inline-c
                      , optparse-applicative >= 0.14.3.0
                      , containers
-  extra-libraries:     stdc++
-                     , c10
-                     , iomp5
-                     , mklml
-                     , caffe2
-                     , torch
-  ghc-options:         -optc-std=c++14 -optc-D_GLIBCXX_USE_CXX11_ABI=0
+                     , ffi
+  ghc-options:       -optc-D_GLIBCXX_USE_CXX11_ABI=0
+ extra-libraries:     torch
```

Comment:
I want to move this extra-lib of torch into ffi-library.
But if I move it, compilation fails.
The message is below.
```
$ stack build ffi
Building all executables for `ffi' once. After a successful build of all of them, only specified executables will be rebuilt.
ffi-0.1.0.0: configure (lib + exe)
Configuring ffi-0.1.0.0...
ffi-0.1.0.0: build (lib + exe)
Preprocessing library for ffi-0.1.0.0..
Building library for ffi-0.1.0.0..

/home/junji-hashimoto/git/hasktorch-dev/hasktorch/ffi-experimental/ffi/src/Aten/Tensor/Cast.hs:2:26: warning:
    -XOverlappingInstances is deprecated: instead use per-instance pragmas OVERLAPPING/OVERLAPPABLE/OVERLAPS
  |
2 |   MultiParamTypeClasses, OverlappingInstances, IncoherentInstances
  |                          ^^^^^^^^^^^^^^^^^^^^

/home/junji-hashimoto/git/hasktorch-dev/hasktorch/ffi-experimental/ffi/src/Aten/Tensor/Implementation.hs:3:26: warning:
    -XOverlappingInstances is deprecated: instead use per-instance pragmas OVERLAPPING/OVERLAPPABLE/OVERLAPS
  |
3 |   MultiParamTypeClasses, OverlappingInstances, TemplateHaskell,
  |                          ^^^^^^^^^^^^^^^^^^^^
[ 1 of 10] Compiling Aten.Tensor.RawType ( src/Aten/Tensor/RawType.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor/RawType.o )
[ 2 of 10] Compiling Aten.Tensor.Interface ( src/Aten/Tensor/Interface.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor/Interface.o )
[ 3 of 10] Compiling Aten.Tensor.FFI  ( .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor/FFI.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor/FFI.o )
[ 4 of 10] Compiling Aten.Tensor.Cast ( src/Aten/Tensor/Cast.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor/Cast.o )
[ 5 of 10] Compiling Aten.Tensor.Implementation ( src/Aten/Tensor/Implementation.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor/Implementation.o )
[ 6 of 10] Compiling Aten.Tensor      ( src/Aten/Tensor.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Tensor.o )
[ 7 of 10] Compiling Aten.Type        ( src/Aten/Type.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Type.o )
[ 8 of 10] Compiling Aten.TH          ( src/Aten/TH.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/TH.o )
[ 9 of 10] Compiling Aten.Native      ( src/Aten/Native.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/Native.o )
[10 of 10] Compiling Aten.NN          ( src/Aten/NN.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/Aten/NN.o )
Preprocessing executable 'ffi-test' for ffi-0.1.0.0..
Building executable 'ffi-test' for ffi-0.1.0.0..
[1 of 1] Compiling Main             ( example/CppTest.hs, .stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/ffi-test/ffi-test-tmp/Main.o )
Key already registered with the same priority: VariableHooks

--  While building package ffi-0.1.0.0 using:
      /home/junji-hashimoto/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.3 --builddir=.stack-work/dist/x86_64-linux/Cabal-2.4.0.1 build lib:ffi exe:ffi-test --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
 
```